### PR TITLE
Remove remaining Swift 5.10 support.

### PR DESCRIPTION
**Pull request by grynspan**
Swift Testing has been accepted into Swift 6. For more information, see [here](https://forums.swift.org/t/accepted-a-new-direction-for-testing-in-swift/72309).

This PR removes all remaining support for the Swift 5.10 toolchain. The package now requires Swift 6.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
Holding until Swift Package Manager is able to support packages requiring Swift 6.

**Comment by grynspan** [comment]
@finestructure Would you mind reviewing this PR? I know there's no SPI-specific changes in it, but _just in case_ you see something here that might cause issues.

**Comment by grynspan** [comment]
@swift-ci please test

**Review by finestructure — APPROVED** [review]

**Comment by grynspan** [comment]
@swift-ci please test

File: Package.swift [hunk 1]
```diff
@@ -121,7 +121,6 @@ extension Array where Element == PackageDescription.SwiftSetting {
       .unsafeFlags(["-require-explicit-sendable"]),
       .enableExperimentalFeature("StrictConcurrency"),
       .enableUpcomingFeature("ExistentialAny"),
-      .enableUpcomingFeature("InferSendableFromCaptures"),
```
[the remark this answers is not in the document]
Comment (grynspan) [reply Package.swift:124]: I didn't actually change this file. I just deleted Package.swift and renamed Package@swift-6.0.swift. So whatever's in there is what we already have and is known-good.

We can refine further in a separate PR.

[the remark this answers is not in the document]
Comment (stmontgomery) [reply Package.swift:124]: The file _was_ changed to `swift-tools-version: 6.0` the top, and I believe that is what will cause the new language mode to be used by default

[the remark this answers is not in the document]
Comment (grynspan) [reply Package.swift:124]: No, it wasn't. `diff` is just confused by the two files.

[the remark this answers is not in the document]
Comment (stmontgomery) [reply Package.swift:124]: As I expected, this change _did_ cause the package to begin using Swift 6 language mode.

**Pull request merged**